test/e2e/user_management_test.go

```diff
@@ -0,0 +1,44 @@
+package e2e
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type userManagementIntegrationTest struct {
+	baseUserIntegrationTest
+}
+
+func TestRunUserManagementIntegrationTest(t *testing.T) {
+	suite.Run(t, &userManagementIntegrationTest{})
+}
+
+func (s *userManagementIntegrationTest) SetupSuite() {
+	userSignupList := &v1alpha1.UserSignupList{}
+	s.testCtx, s.awaitility = testsupport.WaitForDeployments(s.T(), userSignupList)
+	s.hostAwait = s.awaitility.Host()
+	s.namespace = s.awaitility.HostNs
+}
+
+func (s *userManagementIntegrationTest) TearDownTest() {
+	s.testCtx.Cleanup()
+}
+
+func (s *userManagementIntegrationTest) TestUserDeactivation() {
+	s.setApprovalPolicyConfig("automatic")
+
+	userSignup, mur := s.createAndCheckUserSignup(true, "iris-at-redhat-com", "iris@redhat.com", approvedByAdmin()...)
```

Comment:
Please, don't create new user accounts (including all related resources) if it's not really needed - it makes the e2e tests even longer. Please check my comment https://github.com/codeready-toolchain/toolchain-e2e/pull/68#issuecomment-574756595 and consider adding it as a subtest to the main test located in `e2e_test.go`. 

I guess that it would be also good to verify the recreation of resources in case `UserSignup.Deactivated` is reset back to `false`

Reply:
@MatousJobanek we are adding a number of user management-related features currently, see the "User Management Workflows" section at the end of the detailed design doc.  The nature of these features will necessitate the removal of all of a user's resources during testing in order to validate they are functioning correctly, and therefore the creation of new user accounts seems to be unavoidable.  Also since these features are quite comprehensive in nature it makes sense to separate their tests into a separate file - while we just have deactivation implemented so far there are a lot more tests coming shortly.

Reply:
> we are adding a number of user management-related features currently, see the "User Management Workflows" section at the end of the detailed design doc.

I know about that and that's another reason why we should not unnecessarily create resources in the cases when it's not really needed.

> The nature of these features will necessitate the removal of all of a user's resources during testing in order to validate they are functioning correctly, and therefore the creation of new user accounts seems to be unavoidable.

But this can be easily done as part of the already existing test - we don't have to create a special test with all setup for it.

>  Also since these features are quite comprehensive in nature it makes sense to separate their tests into a separate file

These are e2e tests and we should verify the whole e2e flow - that means:
1. provision user account
2. let it working for some time
3. deactivate the user account
4. reactivate 

from my point of view, the best way to simulate the flow is by appending it to the main e2e test. 

> while we just have deactivation implemented so far there are a lot more tests coming shortly.

We all struggle with the e2e tests execution time. There are some Jira issues to improve it and some time ago we were also trying to minimize the number of tests where we provision the user accounts to a necessary minimum - we were merging some tests, deleting the duplicated ones and optimizing the flows. Adding new tests including the whole setup when we can add it to already existing ones is just a step back for me. 

I understand your reasoning - that having a separated test for the functionality looks logical, but this is not a unit test nor an integration test. Unfortunately, the e2e tests take significant time to be completed and we need to keep it as short as possible.

Just to be clear - all that I wrote above is only my point of view and my understanding, but if the team agrees on adding new tests like that, I'm fine with it ;-). WDYT @alexeykazakov @xcoulon @tinakurian

Reply:
Initially I was for creating a separate big user management e2e test which would test all user management related flows. But @MatousJobanek's point of too long test execution is a valid point too.
Adding all the upcoming user tests (diactivaion, disabling, banning, etc) to the existing main e2e test could make it too big and hard to maintain.
On the other hand if creating a dedicated user management test significantly increases the execution time then it's probably not worth separating. But does it? I mean it will be slower for sure but how slower?

Reply:
I'm fine with adding one big user management e2e test with several subtests, but not with adding new several e2e standalone tests for all the user management cases you mentioned. 

As for the times - I made some measuring:
* it takes around 15-17 seconds to set up the test suite 
* up to 2 seconds to create UserSignup & MUR - but I need to point out that in this case, we don't wait for all resources to be provisioned - in other words, we don't wait for `MUR.status == provisioned` (which we should IMO). If we wait for the complete provisioning, then it takes 5-6 seconds
* to tear down the test (delete all resources) it may take up to 4-5 seconds 

The biggest time takes to set up the test-suite. Knowing that we could technically improve the overall execution time by having only one test suite for all tests, but that is for another PR.
 
As for the particular tests - one additional test doesn't bring any significant delay - up to 10 seconds (for provisioning and cleanup). But if we added multiple new standalone tests, then the delay would linearly increase.
From the discussion and also from the code it looks like a that this PR creates a test suite base and in the future, we will add several standalone tests - that is something I'm trying to avoid.

Reply:
> I'm fine with adding one big user management e2e test with several subtests, but not with adding new several e2e standalone tests for all the user management cases you mentioned.

Absolutely. I fully agree that we should not create standalone tests for each user management use case.
I was thinking about having two big test suites:
- the current e2e_test suite for the main flows (it's what we have now)
- a new user management suite with all relevant user management use cases covered.

Both suites are optimized and using as few stand-alone tests as possible. If possible it's just one big e2e flow in each test suite which covers all the use cases.

So, yes, +1 to what you said above @MatousJobanek.

Reply:
Ok, so let's keep it separated, but please the next cases add as subtests to this user-management test.